cmd/create_connector.go

```diff
@@ -21,162 +21,151 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+
 	"github.com/meroxa/cli/utils"
 	"github.com/meroxa/meroxa-go"
 	"github.com/spf13/cobra"
 )
 
-// Config defines a dictionary to be used across our cmd package
-type Config map[string]string
+type CreateConnectorClient interface {
+	GetResourceByName(ctx context.Context, name string) (*meroxa.Resource, error)
+	CreateConnector(ctx context.Context, input meroxa.CreateConnectorInput) (*meroxa.Connector, error)
+}
+
+type CreateConnector struct {
+	input, config, metadata, source, destination, name string
+	pipelineID                                         int
+}
+
+func (cc *CreateConnector) setArgs(args []string) error {
+	if cc.source == "" && cc.destination == "" {
+		return errors.New("requires either a source (--from) or a destination (--to)\n\nUsage:\n  meroxa create connector <custom-connector-name> [--from | --to]")
+	}
+
+	// If user specified an optional connector name
+	if len(args) > 0 {
+		cc.name = args[0]
+	}
 
-// Set a key value pair
-func (c Config) Set(key, value string) error {
-	c[key] = value
 	return nil
 }
 
-// Get a key value pair
-func (c Config) Get(key string) (string, bool) {
-	v, ok := c[key]
-	return v, ok
+func (cc *CreateConnector) setFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVarP(&cc.input, "input", "", "", "command delimited list of input streams")
+	cmd.MarkFlagRequired("input")
+
+	cmd.Flags().StringVarP(&cc.config, "config", "c", "", "connector configuration")
+	cmd.Flags().StringVarP(&cc.metadata, "metadata", "m", "", "connector metadata")
+	cmd.Flags().StringVarP(&cc.source, "from", "", "", "resource name to use as source")
+	cmd.Flags().StringVarP(&cc.destination, "to", "", "", "resource name to use as destination")
+	cmd.Flags().IntVarP(&cc.pipelineID, "pipeline", "", 0, "ID of pipeline to attach connector to") // TODO accept name once we display it in `list connectors` command
```

Comment:
I wonder if we should leave this as `StringVarP` instead and eventually accept both pipeline name and ID once the API supports it.

Reply:
As per our call I changed it so that the command accepts only the pipeline name.